### ISSUE TITLE
test: set and values cannot be used simultaneously

### ISSUE THREAD

**Issue by mederel** [helm-unittest/helm-unittest#124]
I have a reproducer here: 
[helm-unittest-reproducer-global-double-setting.zip](https://github.com/helm-unittest/helm-unittest/files/10853886/helm-unittest-reproducer-global-double-setting.zip)

It boils down to having a required value global.app that's not set by default:

values.yaml

```yaml
global:
  app:
```

template:

```
metadata:
  name: {{ .Values.global.app | required "global.app is mandatory" }}
```

Hence to help out in the writing of tests a requiredValues.yaml is created on the side to set it:

tests/values/requiredValues.yaml

```
global:
  app: my-name
```

But then in a test we write the following:

```yaml
templates:
  - templates/deployment.yaml
tests:
  - it: reproducer test
    values:
      - values/requiredValues.yaml
    set:
      global:
        containerName: blibli
    asserts:
      - equal:
          path: spec.template.spec.containers[0].name
          value: blibli
```

This test fails with: 

```
 FAIL   helm-unittest-reproducer-global-double-setting/tests/deployment_test.yaml
        - reproducer test
                Error: execution error at (helm-unittest-reproducer-global-double-setting/templates/deployment.yaml:4:32): global.app is mandatory
```

It looks like the set `global.containerName` has overridden entirely the requiredValues.yaml setting.

__Important: happens since helm unittest 0.3.0__

**Comment by quintush** [comment]
Hello @mederel,

Thanks for the feedback.

Due to the upgrade of yaml.v3 the merging of the values was failing when the same yaml object is used.
Have added your testcase as an regression-test and fixed the bug.

Will create a new version in a few days (still want to add some other improvements)

Greetings,
@quintush

**Comment by mederel** [comment]
A-MA-ZING, thanks for the quick feedback. Greetings back, @mederel

**Comment by gubarba** [comment]
Thank you for fixing this issue! Looking forward for the the new version release.
Bumped into this migrating from another fork https://github.com/vbehar/helm3-unittest back to this repo.
Also, thank you @quintush for mantaining this as a whole.

**Comment by alxgomz** [comment]
Very same scenario here.
Migrating from helm3-unittest I also faced this issue. Thank you for reviving this repo!

**Comment by rltvty** [comment]
can you please make a new release? I'd like to be able to use this ability.

**Comment by quintush** [comment]
The update is released in 0.3.1.
